### PR TITLE
Khanh/signup error handling

### PR DESCRIPTION
**Pull request by sonyakim-dev**

### CODE REVIEW

File: src/components/UserSignUp.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
```
Comment (sonyakim-dev) [inline src/components/UserSignUp.jsx]: this is an old file. you can remove this

File: src/pages/SignUp/UserSignUp.jsx [hunk 2]
```diff
@@ -10,14 +10,28 @@ const UserSignUp = () => {
   const auth = getAuth();
   const navigate = useNavigate();
 
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
   const handleSignUp = () => {
+    setEmailError('');
+    setPasswordError('');
     createUserWithEmailAndPassword(auth, email.current.value, password.current.value)
```
Comment (sonyakim-dev) [inline src/pages/SignUp/UserSignUp.jsx]: I found a better method for getting email and password without using useRef. Could you change to this?
```js
const handleSignUp = (e) => {
    e.preventDefault();
    const formData = new FormData(e.target);
    const email = formData.get('email');
    const password = formData.get('password');

    setEmailError('');
    setPasswordError('');

    createUserWithEmailAndPassword(auth, email, password)
      .then((userCredential) => {
        navigate('/');
      })
      .catch((error) => {
        const initialMessage = error.code.replaceAll('-', ' ').replace('auth/', '');
        const errorMessage = initialMessage.charAt(0).toUpperCase() + initialMessage.slice(1) + '.';

        if (error.code.includes('password')) {
          setPasswordError(errorMessage);
        } else {
          setEmailError(errorMessage);
        }
      });
  };

  return (
    <section className="max-w-[55rem]">
      <Card className="p-6 my-3">
        <form id="signup-form" onSubmit={handleSignUp}>
          <div className="flex flex-col items-center gap-4">
            <div className="text-xl">Create Account</div>
            <div className="flex w-full flex-col md:flex-row gap-4">
              <Input
                isRequired
                variant="bordered"
                label="Email"
                name="email"
                type="email"
                placeholder="Enter Your Email"
                errorMessage={emailError}
              />
              <Input
                isRequired
                variant="bordered"
                label="Password"
                name="password"
                type="password"
                placeholder="Enter Your Password"
                errorMessage={passwordError}
              />
            </div>
            <Button color="primary" type="submit" name="submit">
              Sign Up
            </Button>
          </div>
        </form>
      </Card>
    </section>
  );
```

**Comment by sonyakim-dev** [comment]
lgtm!!

File: src/components/UserSignUp.jsx (hunk 1 above)
Comment (khanh-v-nguyen) [reply src/components/UserSignUp.jsx]: done!

File: src/pages/SignUp/UserSignUp.jsx (hunk 2 above)
Comment (khanh-v-nguyen) [reply src/pages/SignUp/UserSignUp.jsx]: ok useref has been changed

**Pull request merged**